### PR TITLE
Feature/test

### PR DESCRIPTION
**Pull request by seperalta22**
### Added Features:
- Unit Test for:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - All classes created while implementing the Decorator design pattern.

Participants: @SergioPeralta22 and @Mar12358

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    expect(@book.rentals).to eq([])
+  end
+
+  context 'It should create book and add rentals' do
+    book = Book.new('Ruby is awfull', 'S. P.')
+    it 'should create a new book' do
+      expect(book.title).to eq('Ruby is awfull')
+      expect(book.author).to eq('S. P.')
+    end
+  end
```
Comment (DammyShittu) [inline spec/book_spec.rb:28]: - [ ] Kindly add tests for the `add_rental` method in the Book class. We are required to test the class methods.

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      person = Person.new(1, 'John', 16, parent_permission: true)
+      expect(person.can_use_services?).to be_truthy
+    end
+
+    it 'returns false if person is underage and has no parent permission' do
+      person = Person.new(1, 'John', 16, parent_permission: false)
+      expect(person.can_use_services?).to be_falsey
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/person_spec.rb]: - [ ] Kindly add tests for the `add_rental` method in the Person class.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @SergioPeralta22 and @Mar12358,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frank1738 — APPROVED** [review]
Hi Team , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Good work adding the tests👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**